addons/auth_signup/models/res_users.py

```diff
@@ -238,20 +238,24 @@ def _action_reset_password(self):
                     mail.send()
             _logger.info("Password reset email sent for user <%s> to <%s>", user.login, user.email)
 
-    def send_unregistered_user_reminder(self, after_days=5):
+    def send_unregistered_user_reminder(self, after_days=5, batch_size=100):
         email_template = self.env.ref('auth_signup.mail_template_data_unregistered_users', raise_if_not_found=False)
         if not email_template:
             _logger.warning("Template 'auth_signup.mail_template_data_unregistered_users' was not found. Cannot send reminder notifications.")
             return
         datetime_min = fields.Datetime.today() - relativedelta(days=after_days)
         datetime_max = datetime_min + relativedelta(hours=23, minutes=59, seconds=59)
 
-        res_users_with_details = self.env['res.users'].search_read([
-            ('share', '=', False),
+        domain = [('share', '=', False),
             ('create_uid.email', '!=', False),
             ('create_date', '>=', datetime_min),
             ('create_date', '<=', datetime_max),
-            ('log_ids', '=', False)], ['create_uid', 'name', 'login'])
+            ('log_ids', '=', False)]
+
+        res_users_with_details = self.env['res.users'].search_read(domain, ['create_uid', 'name', 'login'], limit=batch_size)
+        res_users_count = len(res_users_with_details) if len(res_users_with_details) < batch_size else self.env['res.users'].search_count(domain)
+
+        self.env['ir.cron']._notify_progress(0, res_users_count)
```

Comment:
The structure of that code is disturbing to me.  Why do you notify the remaining count here, before processing anything?  Unless there is a commit in the middle (which is not the case here), this remaining count will only be visible at the very end of this transaction.

When there is no intermediate commit, I suggest the following idiom:
```py
records_todo = self.search(domain, limit=batch_size)

# process records_todo here
...

done = len(records_todo)
remaining = 0 if done < batch_size else self.search_count(domain, limit=...)  # limit in case counting them all would be wasteful
return (done, remaining)
```